Makefile

```diff
@@ -1,69 +1,70 @@
-SHELL = /usr/bin/env bash
-OS_NAME = $(shell uname -s)
-BUILD_CONFIGURATION = Debug
-BINARIES_PATH = $(shell pwd)/Binaries
-SRC_PATH = $(shell pwd)/src
-BOOTSTRAP_PATH = $(BINARIES_PATH)/Bootstrap
-BUILD_LOG_PATH =
-HOME_DIR = $(shell cd ~ && pwd)
-DOTNET_VERSION = 1.0.1
-DOTNET = $(BINARIES_PATH)/dotnet-cli/dotnet
-TARGET_FX = netcoreapp1.1
+SHELL := /usr/bin/env bash
+OS_NAME := $(shell uname -s)
+BUILD_CONFIGURATION := Debug
+THIS_MAKEFILE_PATH := $(dir $(realpath $(lastword $(MAKEFILE_LIST))))
+BINARIES_PATH := $(THIS_MAKEFILE_PATH)/Binaries
+SRC_PATH := $(THIS_MAKEFILE_PATH)/src
+BOOTSTRAP_PATH := $(BINARIES_PATH)/Bootstrap
+BUILD_LOG_PATH :=
+DOTNET_VERSION := 1.0.1
+DOTNET_PATH := $(BINARIES_PATH)/dotnet-cli
+DOTNET := $(DOTNET_PATH)/dotnet
+export PATH := $(DOTNET_PATH):$(PATH)
```

Comment:
@agocke Correct - [Make docs](https://www.gnu.org/software/make/manual/html_node/Variables_002fRecursion.html) "To pass down, or export, a variable, make adds the variable and its value to the environment for running each line of the recipe." (those docs are for sub-makes, there might be actual docs for `export`, but it gets the point across)